### ISSUE TITLE
Failed to enable <Group other values in separate bucket> when nested type fields involved

### ISSUE THREAD

**Issue by androidkencai** [ppadovani/KibanaNestedSupportPlugin#47]
**Plugin version**:
[nested-fields-support-6.2.1-1.0.2.zip](https://github.com/ppadovani/KibanaNestedSupportPlugin/releases/download/6.2.1-1.0.2/nested-fields-support-6.2.1-1.0.2.zip)

**Kibana version**:
6.2.1
**Plugins installed**:
KibanaNestedSupportPlugin

**Description of the problem including expected versus actual behavior**:
As title. Also, the problem did not appear If the field is non-nested type. 
**Steps to reproduce**:
 1. New a datatable VIS from UI and add the following configuration in the Editor
Metric - Aggregation: Count
Buckets - Split Rows - Aggregation: Terms
Buckets - Split Rows - Field: <a nested field>

 2. When enabling the check-box of Group other values in separate bucket, the error message below appeared on the UI and failed to group other values in a separate row

**Provide logs and/or server output (if relevant)**:
```
Visualize: Cannot read property 'buckets' of undefined

 Less Info
OK
249s
TypeError: Cannot read property 'buckets' of undefined
    at walkBucketTree (http://localhost:5601/bundles/commons.bundle.js?v=16573:1:342197)
    at buildOtherBucketAgg (http://localhost:5601/bundles/commons.bundle.js?v=16573:1:342468)
    at Object.<anonymous> (http://localhost:5601/bundles/commons.bundle.js?v=16573:129:112364)
    at tryCatch (http://localhost:5601/bundles/vendors.bundle.js?v=16573:70:99706)
    at Generator._invoke (http://localhost:5601/bundles/vendors.bundle.js?v=16573:70:101781)
    at Generator.prototype.(anonymous function) [as next] (http://localhost:5601/bundles/vendors.bundle.js?v=16573:70:99990)
    at step (http://localhost:5601/bundles/commons.bundle.js?v=16573:129:109255)
    at http://localhost:5601/bundles/commons.bundle.js?v=16573:129:109458
    at new Promise (<anonymous>)
    at BucketAggType.<anonymous> (http://localhost:5601/bundles/commons.bundle.js?v=16573:129:109174)
```

**Comment by ppadovani** [comment]
Thanks for the report, I'll take a look at this in the next day or so.

**Comment by ppadovani** [comment]
Ok... I see the issue, there is a file I missed that requires an override to unwrap the outer nested egg response. Instead of a quick fix, I'm going to see if I can strip the nested agg wrappers from the response before being processed, it would allow me to override less code.

**Comment by ppadovani** [comment]
Quick update... fixed a part of the issue.. but found that there is a modularization of filter code that is going to have to be overridden to understand nested fields. This will take a few days of work to ensure I get it right.

**Comment by ppadovani** [comment]
Fixed and released 6.2.1-1.0.3
